src/GovUk.Education.ExploreEducationStatistics.Admin/Requests/DataBlockRequests.cs

```diff
@@ -26,7 +26,9 @@ public class Validator : AbstractValidator<DataBlockCreateRequest>
         public Validator()
         {
             RuleFor(request => request.Heading)
-                .NotEmpty();
+                .NotEmpty()
+                .MaximumLength(120)
+                .WithMessage("Table title must be 120 characters or less");
```

Comment:
I would remove all of extra `.WithMessage` configuration that's been added throughout this PR and just rely on the default FluentValidation error messages.

From an API point of view, FluentValidation  already does a decent job of describing the error, both in a summary format and a detailed error format which includes additional details like the min length, max length, and the value provided.

The error response is also self describing, containing a path to the field in error state, in this case `heading`, which is more accurate from an API point of view than `Table title`. As far as I know these BE errors don't get presented to users so there's no need to provide friendlier names for each of the fields.

The maximum length in the out of the box generated message is also guaranteed to be accurate to the actual maximum length, which is not the case if you define a custom message where the two values could get out of sync.

Reply:
Ok yeah I see your reasoning and I expect that the mismatch between names in request fields/error paths and the front end names for those fields is quite a widespread issue.

I tried it out to see what the auto generated message for `MaximumLength(120)` would look like without providing a custom message, and in the API response it will be something like

```json
"message": "Must be 120 characters or fewer (was 123)."
```

Interestingly when I reduced the max length for the data block title right down to 5 (on the backend only) to deliberately cause a backend validation error this wasn't handled by the frontend. The 400 Bad Request with validation error caused the Sorry... page on the frontend.

I'm guessing you were expecting this to be handled, but this might be due to the mismatching between the names?

Reply:
It's not handled properly in _every_ bit of the frontend. In some cases you'll see the error in exactly the same way as you would if it was client-side validation, but in others you will only see the detailed error messages if you check the browser console (i.e. by expanding the Axios error details, or in the network tab). For example, in cases where a _list_ of errors can be returned for a single field (such as the bulk zip upload), only the first error is shown to the client, but the rest can be viewed in the console - this is a limitation of the validation summary component.

FYI I tested the backend validation by _disabling_ the frontend rule, rather than altering it. I wonder if there's an issue on the frontend whereby it fails to update the "valid" state upon receiving an "invalid" response? 🤷

Reply:
I chatted with Nick about this as he knows a bit more about how validation is coordinated across the frontend and backend, and we agreed there shouldn't be a need for defining custom error messages on the backend for things like max length validation on a field-by-field basis. We'll just end up defining the same message over and over again, as can be seen in your new `ValidationConstants.cs` file where you've essentially got the same max length message repeated 7 times. If we accept this change it would also raise the question of where else we've already got length validation on the backend which would now need these type of messages adding in to make those consistent. Presumably there would be many places we'd need to do this, not just those covered by this change.

I think the file uploads may work differently because in those cases you've got no frontend form object or backend request type) and there's a multipart form upload instead, and that's probably being handled differently where an error from the backend may end up being presented to the user.

In the other cases though the frontend should be able to handle the backend errors by using an error mapper to map which fields the server-side validation errors correspond to.

So something like this should work

```javascript
const errorMappings = [
  mapFieldErrors<FormValues>({
    target: 'heading',
    messages: {
      MaximumLength: 'Title must be 120 characters or fewer',
    },
  }),
];
```

The `errorMappings` is then set as a property of `FormProvider`.

A backend validation error would be as follows:

```json
    "errors": [
        {
            "message": "Must be 120 characters or fewer (was 123).",
            "path": "heading",
            "code": "MaximumLength",
            "detail": {
                "minLength": 0,
                "maxLength": 120,
                "totalLength": 123,
                "value": "My very long title exceeding 120 chars!"
            }
        }
    ]
```

Using the frontend `errorMappings ` code above, the frontend should then remap the message for `MaximumLength` to its own message. This gives us the ability for the frontend to own all the messages which the user ends up seeing (`Title must be 120 characters or fewer` in my example), and we don't really care what the backend validation message is.

From the back end API's point of view the out-of-the-box FluentValidation message is sufficient and we don't get bogged down with the wording of messages. The front end massages can be written in a way that read better for users and are consistent with the GOV.UK style.

In future the frontend error mapping mechanism could be improved to automatically map common backend errors like `Required` or `MaximumLength` and extract certain details from the `detail` element like `detail.minLength` and `detail.maxLength` and substitute them into a message. This will mean we'll only have to define the max length  message in one place on the frontend and get a consistent message everywhere.

Reply:
Suggested changes failed due to FE prioritising server errors in serverValidation.ts (ln. 186) so I've removed the BE messaging.
![image](https://github.com/user-attachments/assets/3d5f38f1-d81d-41cb-b78e-c0376e308ec4)